geode-wan/src/distributedTest/java/org/apache/geode/internal/cache/wan/parallel/ParallelGatewaySenderOperationsDUnitTest.java

```diff
@@ -1514,10 +1595,34 @@ private void waitForSendersRunning() {
     vm7.invoke(() -> waitForSenderRunningState("ln"));
   }
 
+  private void waitForAllSendersNotRunning() {
+    vm4.invoke(() -> waitForSenderNonRunningState("ln"));
+    vm5.invoke(() -> waitForSenderNonRunningState("ln"));
+    vm6.invoke(() -> waitForSenderNonRunningState("ln"));
+    vm7.invoke(() -> waitForSenderNonRunningState("ln"));
+  }
+
   private void validateParallelSenderQueueAllBucketsDrained() {
     vm4.invoke(() -> validateParallelSenderQueueAllBucketsDrained("ln"));
     vm5.invoke(() -> validateParallelSenderQueueAllBucketsDrained("ln"));
     vm6.invoke(() -> validateParallelSenderQueueAllBucketsDrained("ln"));
     vm7.invoke(() -> validateParallelSenderQueueAllBucketsDrained("ln"));
   }
+
+  private static class CountSentPQRMObserver extends DistributionMessageObserver {
+    private int numberOfSentPQRM = 0;
```

Comment:
I'm not sure if this needs to be thread safe or not, but you should use `AtomicInteger` just to be safe:
```
private final AtomicInteger numberOfSentPQRM = new AtomicInteger();
```
```
if (message instanceof ParallelQueueRemovalMessage) {
  numberOfSentPQRM.addAndGet(message.getRecipients().size();
}
```

Reply:
Updated